titus-api/src/main/java/com/netflix/titus/api/FeatureRolloutPlans.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.api;
+
+import com.netflix.titus.common.util.feature.FeatureRollout;
+
+public interface FeatureRolloutPlans {
+    @FeatureRollout(
+            featureId = "securityGroupsRequired",
+            deadline = "03/30/2019",
+            description = "Require all clients to provide the security group in the job descriptor."
+    )
+    String SECURITY_GROUPS_REQUIRED_FEATURE = "securityGroupsRequired";
+
+    @FeatureRollout(
+            featureId = "iamRoleRequired",
+            deadline = "03/30/2019",
+            description = "Require all clients to provide the IAM role in the job descriptor."
+    )
+    String IAM_ROLE_REQUIRED_FEATURE = "iamRoleRequired";
+
+    @FeatureRollout(
+            featureId = "entryPointStrictValidation",
+            deadline = "03/30/2019",
```

Comment:
removing this requires changes to upstream systems (Spinnaker, Meson, ...) so they stop sending entrypoints as flat strings to us. This is not realistically changing in Q1/2019.

TBH, there is a chance we will never be able to remove "support" for this legacy usage, at least not until we break backwards compatibility with a new v4 API, or we add a whitelist of callers that are allowed to use flat string entrypoints.

Can we disable the deadline for this one then? Or does it make sense to not track it as a `FeatureRollout`?

If not, let's set the deadline to say jun/2019, and evaluate blocking usage for most callers (only allowing a few select callers) of flat string entrypoints.

Reply:
I am ok with setting the deadline to latter time. I have changed this to: 06/30/2019